### ISSUE TITLE
Make unspecified-encoding work for pathlib.Path.write_text()

### ISSUE THREAD

**Issue by trym-b** [PyCQA/pylint#4945]
### Current problem

I very much like `unspecified-encoding` for enforcing the encoding parameter when using `open`, however it does not seem to work with the more modern way of writing to files through `pathlib`, which also has an encoding argument.

Here is an example
```python
from pathlib import Path


def open_test() -> None:
    with open("some_file") as open_file:
        print(open_file.read())


def read_text_test() -> None:
    print(Path("some_file").read_text())

```

When running the following command, only the `open` call without encoding will be detected by pylint.
```
$ pylint ./test_pylint_pathlib.py --disable missing-module-docstring --disable missing-function-docstring
************* Module test_pylint_pathlib
test_pylint_pathlib.py:5:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)

------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 8.33/10, +0.00)

```

### Desired solution

I want calls to [`pathlib.Path.write_text()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text) / [`pathlib.Path.read_text()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text) to also be detected by pylint.

Basically something like this as output of the code above:
```
$ pylint ./test_pylint_pathlib.py --disable missing-module-docstring --disable missing-function-docstring
************* Module test_pylint_pathlib
test_pylint_pathlib.py:5:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
test_pylint_pathlib.py:10:28: W1514: Using read_text without explicitly specifying an encoding (unspecified-encoding)

------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 8.33/10, +0.00)

```
